test/grantsrole_test.js

```diff
@@ -0,0 +1,103 @@
+suite("ScopeResolver (grantsRole)", () => {
+  let ScopeResolver = require('../auth/scoperesolver');
+  let assert        = require('assert');
+
+  // Test cases for grantsRole
+  [
+    {
+      // cases with *
+      scope:    '*',
+      role:     '*',
+      result:   true
+    }, {
+      scope:    '*',
+      role:     'client-id:queue',
+      result:   true
+    }, {
+      scope:    '*',
+      role:     'task-run-id:*',
+      result:   true
+    }, {
+      // cases with as*
+      scope:    'as*',
+      role:     '*',
+      result:   true
+    }, {
+      scope:    'as*',
+      role:     'client-id:queue',
+      result:   true
+    }, {
+      scope:    'as*',
+      role:     'task-run-id:*',
+      result:   true
+    }, {
+      // cases with assume:*
+      scope:    'assume:*',
+      role:     'client-id:queue',
+      result:   true
+    }, {
+      scope:    'assume:*',
+      role:     'task-run-id:*',
+      result:   true
+    }, {
+      scope:    'assume:*',
+      role:     '*',
+      result:   true
+    }, {
+      // cases with assume:<prefix>*
+      scope:    'assume:client-id:*',
+      role:     'client-id:queue',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:*',
+      role:     'task-run-id:*',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:*',
+      role:     '*',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:*',
+      role:     'task-run-*',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:*',
+      role:     'client-id:queue',
+      result:   false
+    }, {
+      scope:    'assume:task-run-id:*',
+      role:     'client-id:*',
+      result:   false
+    }, {
+      // cases with assume:roleId
+      scope:    'assume:client-id:queue',
+      role:     'client-id:queue',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:12345',
+      role:     'task-run-id:72345',
+      result:   false
+    }, {
+      scope:    'assume:task-run-id:12345',
+      role:     'task-run-id:*',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:12345',
+      role:     '*',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:12345',
+      role:     'task-run-*',
+      result:   true
+    }, {
+      scope:    'assume:task-run-id:12345',
+      role:     'client-id:*',
+      result:   false
+    },
+  ].forEach(({scope, role, result}) => {
+    test(`grantsRole(${scope}, ${role}) === ${result}`, () => {
+      assert(ScopeResolver.grantsRole(scope, role) === result,
+             `Expected grantsRole(${scope}, ${role}) === ${result}`);;
+    });
+  });
+});
```

Comment:
there is some in the `role_test.js` mainly done so because I prefer testing everything at once...

Otherwise it's a lot of code adding things entities, etc... The role_test also hits many of the reload points... 

But I probably need a test for recursive roles, normal and prefix roles are already covered...